ws-nextjs-app/pages/_document.page.tsx

```diff
@@ -24,11 +34,30 @@ type DocProps = {
 export default class AppDocument extends Document<DocProps> {
   static async getInitialProps(ctx: DocumentContext) {
     const initialProps = await Document.getInitialProps(ctx);
-    const isApp = isAppPath(ctx.asPath || '');
+    const path = ctx.asPath || '';
+    const isApp = isAppPath(path);
 
     // Read env variables from the server and expose them to the client
     const clientSideEnvVariables = getProcessEnvAppVariables();
 
+    if (ctx.res?.statusCode === OK) {
+      logger.debug(SERVER_SIDE_RENDER_REQUEST_RECEIVED, {
+        url: path,
+        headers: removeSensitiveHeaders(ctx.req?.headers),
+        pageType: derivePageType(path),
+      });
+    }
+
+    if (ctx.res?.statusCode === INTERNAL_SERVER_ERROR) {
+      logger.error(SERVER_SIDE_REQUEST_FAILED, {
+        status: INTERNAL_SERVER_ERROR,
+        message: ctx.res?.statusMessage,
+        url: path,
+        headers: removeSensitiveHeaders(ctx.req?.headers),
+        pageType: derivePageType(path),
+      });
+    }
```

Comment:
Should we calculate the headers & pageType once, and then pass it into the logger?

 ```suggestion
    const headers = removeSensitiveHeaders(ctx.req?.headers);
    const pageType = derivePageType(path);
 
    if (ctx.res?.statusCode === OK) {
      logger.debug(SERVER_SIDE_RENDER_REQUEST_RECEIVED, {
        url: path,
        headers,
        pageType,
      });
    }

    if (ctx.res?.statusCode === INTERNAL_SERVER_ERROR) {
      logger.error(SERVER_SIDE_REQUEST_FAILED, {
        status: INTERNAL_SERVER_ERROR,
        message: ctx.res?.statusMessage,
        url: path,
        headers,
        pageType,
      });
    }
```

Reply:
Yes! Thanks.